pkg/snapshot/filesystem.go

```diff
@@ -0,0 +1,782 @@
+					},
+					Key: "MINIO_SECRET_KEY",
+				},
+			},
+		},
+	}
+
+	return &appsv1.Deployment{
+		TypeMeta: metav1.TypeMeta{
+			APIVersion: "apps/v1",
+			Kind:       "Deployment",
+		},
+		ObjectMeta: metav1.ObjectMeta{
+			Name: FileSystemMinioDeploymentName,
+		},
+		Spec: appsv1.DeploymentSpec{
+			Replicas: pointer.Int32Ptr(1),
+			Selector: &metav1.LabelSelector{
+				MatchLabels: map[string]string{
+					"app": "kotsadm-fs-minio",
+				},
+			},
+			Strategy: appsv1.DeploymentStrategy{
+				Type: appsv1.RecreateDeploymentStrategyType,
+			},
+			Template: corev1.PodTemplateSpec{
+				ObjectMeta: metav1.ObjectMeta{
+					Labels: map[string]string{
+						"app": "kotsadm-fs-minio",
+					},
+					Annotations: map[string]string{
+						"kots.io/fs-minio-creds-secret-checksum": secretChecksum,
+					},
+				},
+				Spec: corev1.PodSpec{
+					SecurityContext:  &securityContext,
+					ImagePullSecrets: imagePullSecrets,
+					Containers: []corev1.Container{
+						{
+							Image:           image,
+							ImagePullPolicy: corev1.PullIfNotPresent,
+							Name:            "minio",
+							Ports: []corev1.ContainerPort{
+								{Name: "http", ContainerPort: 9000},
+							},
+							Env: env,
+							VolumeMounts: []corev1.VolumeMount{
+								{
+									Name: "data", MountPath: "/data",
+								},
+							},
+							Args: []string{"--quiet", "server", "data"},
+							LivenessProbe: &corev1.Probe{
+								Handler: corev1.Handler{
+									HTTPGet: &corev1.HTTPGetAction{
+										Path: "/minio/health/live",
+										Port: intstr.FromInt(9000),
+									},
+								},
+								InitialDelaySeconds: 5,
+								PeriodSeconds:       20,
+							},
+							ReadinessProbe: &corev1.Probe{
+								Handler: corev1.Handler{
+									HTTPGet: &corev1.HTTPGetAction{
+										Path: "/minio/health/ready",
+										Port: intstr.FromInt(9000),
+									},
+								},
+								InitialDelaySeconds: 5,
+								PeriodSeconds:       20,
+							},
+						},
+					},
+					Volumes: []corev1.Volume{
+						{
+							Name:         "data",
+							VolumeSource: volumeSourceFromFileSystemConfig(deployOptions.FileSystemConfig),
+						},
+					},
+				},
+			},
+		},
+	}, nil
+}
+
+func updateFileSystemMinioDeployment(existingDeployment, desiredDeployment *appsv1.Deployment) *appsv1.Deployment {
+	if len(existingDeployment.Spec.Template.Spec.Containers) == 0 {
+		// hmm
+		return desiredDeployment
+	}
+
+	existingDeployment.Spec.Replicas = desiredDeployment.Spec.Replicas
+
+	if existingDeployment.Spec.Template.Annotations == nil {
+		existingDeployment.Spec.Template.ObjectMeta.Annotations = map[string]string{}
+	}
+	existingDeployment.Spec.Template.ObjectMeta.Annotations["kots.io/fs-minio-creds-secret-checksum"] = desiredDeployment.Spec.Template.ObjectMeta.Annotations["kots.io/fs-minio-creds-secret-checksum"]
+
+	existingDeployment.Spec.Template.Spec.Containers[0].Image = desiredDeployment.Spec.Template.Spec.Containers[0].Image
```

Comment:
Maybe check container name is `"minio"` in case we add more containers?   That way we can catch this error eralier.